### PR TITLE
Refactor BuildFilterConditions: add search_word filter

### PR DESCRIPTION
**Pull request by zheng1**
- add `db.Like` operator to db.conditions
- add `search_word` filter to manager.BuildFilterConditions

### CODE REVIEW

File: pkg/manager/common.go [hunk 1]
```diff
@@ -23,36 +25,82 @@ type Request interface {
...
+	return nil
+}
 
-func BuildFilterConditions(req Request, columns ...string) dbr.Builder {
-	var filter []dbr.Builder
+func getStringValue(param interface{}) interface{} {
+	switch value := param.(type) {
+	case string:
+		if value == "" {
+			return nil
```
[the remark this answers is not in the document]
Comment (zheng1) [reply pkg/manager/common.go]: In this func, return value is `interface{}` and `nil` means return value is invalid. There is three possible return values: 1. nil 2. string 3. []string.

File: pkg/manager/common.go [hunk 2]
```diff
@@ -23,36 +25,82 @@ type Request interface {
...
+		}
+		if len(values) == 0 {
+			return nil
+		}
+		return values
+	}
+	return nil
+}
+
+func BuildFilterConditions(req Request, tableName string, exclude ...string) dbr.Builder {
```
[the remark this answers is not in the document]
Comment (zheng1) [reply pkg/manager/common.go]: For displayColumns we'll implement later, but not in this func. Like qingcloud, this func only deals with the filter conditions.

File: pkg/models/common.go [hunk 3]
```diff
@@ -10,6 +10,38 @@ import (
 	"openpitrix.io/openpitrix/pkg/utils"
 )
 
+var SearchWordColumnTable = []string{
+	AppTableName,
+}
+
+// columns that can be search through sql 'like' operator
```
[the remark this answers is not in the document]
Comment (zheng1) [reply pkg/models/common.go]: I will fix it.

File: pkg/manager/common.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chilianyi) [reply pkg/manager/common.go]: OK, the method name may not very good.

**Pull request merged**